openedx/openedx-platform
# SE-1773 Logs failed heartbeat checks

Adds error logging for failed heartbeat checks, to make it possible to diagnose issues on monitored servers.

**JIRA tickets**: SE-1773, [OSPR-3904](https://openedx.atlassian.net/browse/OSPR-3904)

**Sandbox URL**: TBD - sandbox is being provisioned.

**Merge deadline**: None

**Testing instructions**:

1. Deploy this change to the open edx devstack.
1. Stop the mongo container to trigger an error: 
   ```bash
   (devstack) devstack $  docker-compose stop  mongo
   ```
1. Visit your LMS's `/heartbeat?extended` URL, while running `make lms-logs`.
   Note the heartbeat error is logged when the page returns 503, e.g.
   ```
   2019-10-28 07:33:23,969 ERROR 5158 [openedx.core.djangoapps.heartbeat.views] [user None] views.py:36 - Heartbeat check failed (503): {'modulestore': {'status': False, 'message': u"Can't connect to edxapp"}, 'celery': {'status': True, 'message': u"{'time': '0.000463008880615'}"}, 'sql': {'status': True, 'message': u'OK'}}
   ```
1. Restart the mongo container to demonstrate normal operations.
   ```bash
   (devstack) devstack $  docker-compose up -d mongo
   ```
1. Re-visit `/heartbeat?extended`, and note that no error is logged, and service returns 200.

**Reviewers**

### Please consider the following when opening a pull request:

- Link to the relevant JIRA ticket(s) and tag any relevant team(s).
- Squash your changes down into one or more discrete commits.
  In each commit, include description that could help a developer
  several months from now.
- If running `make upgrade`, run _as close to the time of merging as possible_
  to avoid accidentally downgrading someone else's package.
  Put the output of `make upgrade` in its own separate commit,
  decoupled from other code changes.
- Aim for comprehensive test coverage, but remember that
  automated testing isn't a substitute for manual verification.
- Carefully consider naming, code organization, dependencies when adding new code.
  Code that is amenable to refactoring and improvement benefits all platform developers,
  especially given the size and scope of edx-platform.
  Consult existing Architectural Decision Records (ADRs),
  including those concerning the app(s) you are changing and
  [those concerning edx-platform as a whole](https://github.com/edx/edx-platform/tree/master/docs/decisions).
